### ISSUE TITLE
Add in-place ops. 

### ISSUE THREAD

**Issue by ry** [propelml/propel#272]
Follow the PyTorch convention of using underscores to denote in-place.
For example:

`x.add(y)` will return a new tensor, with neither `x` nor `y` updated.

`x.add_(y)` will return the sum, with `x` updated to the sum.

**Comment by ryanmurakami** [comment]
I'd like to work on this, if that's okay.

**Comment by ry** [comment]
@ryanmurakami Yes, you're welcome to do so! Please start small with just `add_()` so that we can check that the structure is okay before doing all of the other ops.
